### PR TITLE
Add tests for Allowinsecureredirect parameter in Web cmdlets

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Some tests for -AllowInsecureRedirect, they might need some small changes

## PR Context
#18902 #18546 #18595

### CODE REVIEW

**Comment by iSazonov** [comment]
@CarloToso Please look test fails.

**Comment by CarloToso** [comment]
@iSazonov fixed tests

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 1]
```diff
@@ -937,6 +957,25 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
...
+            $httpUri = Get-WebListenerUrl -Test 'Get'
+            $uri = Get-WebListenerUrl -Test 'Redirect' -Https -Query @{destination = $httpUri}
+            $command = "Invoke-WebRequest -Uri '$uri' -SkipCertificateCheck -AllowInsecureRedirect"
+
+            $result = ExecuteWebCommand -command $command
+            $jsonContent = $result.Output.Content | ConvertFrom-Json
+            $jsonContent.headers.Host | Should -Match $httpUri.Authority
+        }
+
+        It "Validate Invoke-WebRequest Https to Http redirect fails" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:971]: Here and below:
```suggestion
        It "Validate Invoke-WebRequest Https to Http redirect without -AllowInsecureRedirect" {
```

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@CarloToso Great! Thanks!

**Pull request merged**